### ISSUE TITLE
[REVIEW]: rf: Receiver function calculation in seismology

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1808]
**Submitting author:** @trichter (<a href="http://orcid.org/0000-0002-8378-559X">Tom Eulenfeld</a>)
**Repository:** <a href="https://github.com/trichter/rf" target ="_blank">https://github.com/trichter/rf</a>
**Version:** 0.9.1
**Editor:** @leouieda
**Reviewers:** @ThomasLecocq, @seisman
**Archive:** <a href="https://doi.org/10.5281/zenodo.3741366" target="_blank">10.5281/zenodo.3741366</a>

## Status

[![status](https://joss.theoj.org/papers/47f72fe33d4defd396270bba1e0a8a58/status.svg)](https://joss.theoj.org/papers/47f72fe33d4defd396270bba1e0a8a58)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/47f72fe33d4defd396270bba1e0a8a58"><img src="https://joss.theoj.org/papers/47f72fe33d4defd396270bba1e0a8a58/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/47f72fe33d4defd396270bba1e0a8a58/status.svg)](https://joss.theoj.org/papers/47f72fe33d4defd396270bba1e0a8a58)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/trichter/rf" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@ThomasLecocq & @seisman, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @leouieda know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/trichter/rf">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@trichter) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/trichter/rf">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@trichter) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @ThomasLecocq, @seisman it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01808/joss.01808/10.21105.joss.01808.pdf)

**Comment by leouieda** [comment]
👋🏼 @trichter @seisman @ThomasLecocq  this is the review thread for the paper. All of our communications will happen here from now on.

Both reviewers have checklists at the top of this thread with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention `openjournals/joss-reviews#1808` so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

We aim for reviews to be completed within about 4 weeks. Please let me know if any of you require some more time. We can also use Whedon (our bot) to set automatic reminders if you know you'll be away for a known period of time.

Please feel free to ping me here (@leouieda) or email me privately if you have any questions/concerns.

**Comment by leouieda** [comment]
:wave: Hi @seisman @ThomasLecocq just checking in on the review progress. Are there any updates?

**Comment by seisman** [comment]
@leouieda I should be able to finish the review before this weekend.

**Comment by seisman** [comment]
@leouieda I've finished my review and my comments have been addressed. The submission now looks good to me.

**Comment by trichter** [comment]
@seisman Thank you very much for the review!

**Comment by leouieda** [comment]
:wave: Hi @ThomasLecocq just checking in to see how the review if progressing. Let me know if I can help in any way.

**Comment by trichter** [comment]
I updated the reference for the telewavesim package, which was recently published in JOSS.

**Comment by trichter** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01808/joss.01808/10.21105.joss.01808.pdf)

**Comment by leouieda** [comment]
Hi @trichter thanks for the update. Apologies for the delay in this submission. I'll see if we can speed things up after AGU next week.

**Comment by leouieda** [comment]
Waiting on a reply from @ThomasLecocq. If we don't hear back, I'll start looking for another reviewer (or probably just assign myself as a reviewer). Thanks for your patience @trichter :slightly_smiling_face:

**Comment by ThomasLecocq** [comment]
wow guyz, sorry for this, noticed I disabled the github notifications a while ago... and @leouieda your emails went to spam... so I masterfully failed my assignment here...

Will do this week. Sorry about this.

**Comment by leouieda** [comment]
@ThomasLecocq any updates on this review?

**Comment by ThomasLecocq** [comment]
All checks above are OK for me, I just proposed some changes to make windows fortran a real thing :)

(I cannot change the checkboxes above, sorry)

and sorry for the awful delay....

**Comment by leouieda** [comment]
@whedon add @ThomasLecocq as reviewer

**Comment by whedon** [comment]
OK, @ThomasLecocq is now a reviewer

**Comment by leouieda** [comment]
Hi @ThomasLecocq thank you for the review and comments! I tried re-assigning you. Please check your inbox for an invitation to the organization so that you can edit the checkboxes. You might want to "unwatch" this repository since there will be a lot of activity.

**Comment by ThomasLecocq** [comment]
@leouieda all OK, I can now check the boxes

**Comment by ThomasLecocq** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

# Ask Whedon to check repository statistics for the submitted software
@whedon check repository

```

**Comment by ThomasLecocq** [comment]
@leouieda all seem green light to me!

**Comment by trichter** [comment]
@ThomasLecocq Thank you for the review, I am glad that you found the time.
@leouieda Anything I can do to move forward the status of this manuscript?

**Comment by arfon** [comment]
Dear authors and reviewers

We wanted to notify you that in light of the current COVID-19 pandemic, JOSS has decided to suspend submission of new manuscripts and to handle existing manuscripts (such as this one) on a "best efforts basis". We understand that you may need to attend to more pressing issues than completing a review or updating a repository in response to a review. If this is the case, a quick note indicating that you need to put a "pause" on your involvement with a review would be appreciated but is not required.

Thanks in advance for your understanding.

_Arfon Smith, Editor in Chief, on behalf of the JOSS editorial team._

**Comment by leouieda** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1088/1749-4699/8/1/014003 is OK
- 10.1504/IJCSE.2009.029165 is OK
- 10.21105/joss.01818 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by leouieda** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01808/joss.01808/10.21105.joss.01808.pdf)

**Comment by leouieda** [comment]
@trichter I made a few editorial suggestions to the paper. After those are integrated, there a couple of things that you'll need to do before acceptance:

- Double check authors and affiliations (including ORCIDs)
- Make a release of the software with the latest changes from the review and **post the version number here**. This is the version that will be used in the JOSS paper.
- Archive the release on Zenodo/figshare/etc and **post the DOI here**.
- Make sure that the title and author list (including ORCIDs) in the archive match those in the JOSS paper.

**Comment by trichter** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01808/joss.01808/10.21105.joss.01808.pdf)

**Comment by trichter** [comment]
I worked through your posted list:

* Added your editorial suggestions and my orcid to the paper
* Published a new version ``0.9.1``
* Uploaded it to zenodo, the doi is ``10.5281/zenodo.3741366``

**Comment by leouieda** [comment]
@whedon set 0.9.1 as version

**Comment by whedon** [comment]
OK. 0.9.1 is the version.

**Comment by leouieda** [comment]
@whedon set 10.5281/zenodo.3741366 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3741366" target="_blank">10.5281/zenodo.3741366</a> is the archive.

**Comment by leouieda** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01808/joss.01808/10.21105.joss.01808.pdf)

**Comment by leouieda** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1088/1749-4699/8/1/014003 is OK
- 10.1504/IJCSE.2009.029165 is OK
- 10.21105/joss.01818 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1408

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1408, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by leouieda** [comment]
@trichter thank you for implementing these final fixes. It's my pleasure to recommend your submission for publication in JOSS :tada: 

A JOSS Associate Editor-in-Chief will stop by and do the final acceptance and publication (usually quick but please be patient given the current global crisis).

@seisman @ThomasLecocq thank for your reviews of this submission!

**Comment by danielskatz** [comment]
Thanks @leouieda - I'll take this now

**Comment by danielskatz** [comment]
@trichter - there are some small changes in https://github.com/trichter/rf/pull/27

**Comment by trichter** [comment]
@danielskatz Thanks for these. I merged your branch.

**Comment by trichter** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01808/joss.01808/10.21105.joss.01808.pdf)

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1247163382677082113) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1409
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01808](https://doi.org/10.21105/joss.01808)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by danielskatz** [comment]
Thanks to @ThomasLecocq & @seisman for reviewing, and @leouieda for editing!

Congratulations to @trichter!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01808/status.svg)](https://doi.org/10.21105/joss.01808)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01808">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01808/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01808/status.svg
   :target: https://doi.org/10.21105/joss.01808
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01808/status.svg)](https://doi.org/10.21105/joss.01808)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
